### PR TITLE
paddle quantile function #10932 added

### PR DESCRIPTION
**Pull request by mosesdaudu001**
- [quantile] #10932 

hi @MahmoudAshraf97, pls you could check this link to see how I tested the quantile function and confirmed that it was accurate with other backends (jax, numpy, tensorflow and pytorch) using their corresponding backend quantile functions that I got from ivy.

https://colab.research.google.com/drive/1YQdibuuy5AysRHM_1lIE7DtZXoQvDwNw?usp=share_link

### CODE REVIEW

**Review by MahmoudAshraf97 — CHANGES_REQUESTED** [review]
Hi @mosesdaudu001 and thanks for the contribution and the awesome test notebook, there are some changes needed and we're good to merge:
these are the input data:
```
a = [[10., 7., 4.], [3., 2., 1.]]
q = 0.5
```
1. when `axis=None`, paddle function behaves as if `axis=0` which is not the case with other backends
2. the `keepdims` argument is not working as intended in paddle, 
`quantile(data_tf, q, axis=1)`
`>>>[7., 2.]` n all backends with `shape=(2,)` while in paddle the `shape=(2, 1)`
3. when `q` is a paddle tensor, the result is different than passing it a a float

these are all the test cases:
```
a = [[10., 7., 4.], [3., 2., 1.]]
q = 0.5
>>> ivy.quantile(a, q)
>>> ivy.array(3.5)


>>> ivy.quantile(a, q, axis=0)
>>> ivy.array([6.5, 4.5, 2.5])


>>> ivy.quantile(a, q, axis=1)
>>>ivy.array([7.,  2.])


>>> ivy.quantile(a, q, axis=1, keepdims=True)
>>>ivy.array([[7.],[2.]])

a = [1., 2., 3., 4.]
q = [0.3, 0.7]
>>> ivy.quantile(a, q, interpolation='lower')
>>> ivy.array([1., 3.])
```

**Comment by MahmoudAshraf97** [comment]
Hi, Please open a new PR after forking from the new PaddlePaddle branch, sorry for the inconvenience
